SingularityUI/app/components/requestForm/RequestForm.jsx

```diff
@@ -0,0 +1,564 @@
+      return false;
+    } else if (FIELDS_BY_REQUEST_TYPE[this.getValue('requestType')].indexOf(fieldId) === -1) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  getButtonsDisabled(type) {
+    if (this.isEditing() && this.getValue('requestType') !== type) {
+      return 'disabled';
+    } else {
+      return null;
+    }
+  }
+
+  updateField(fieldId, newValue) {
+    this.props.update(FORM_ID, fieldId, newValue);
+  }
+
+  updateTypeButtonClick(event) {
+    event.preventDefault();
+    this.updateField('requestType', event.target.value);
+  }
+
+  getScheduleType() {
+    if (this.isEditing() && !(this.props.form && this.props.form.scheduleType)) {
+      if (this.props.request.request[QUARTZ_SCHEDULE]) {
+        return QUARTZ_SCHEDULE;
+      } else {
+        return CRON_SCHEDULE;
+      }
+    } else {
+      if (this.props.form && this.props.form.scheduleType) {
+        return this.props.form.scheduleType;
+      } else {
+        return CRON_SCHEDULE;
+      }
+    }
+  }
+
+  renderRequestTypeSelectors() {
+    const tooltip = (
+      <ToolTip id="cannotChangeRequestTypeAfterCreation">Option cannot be altered after creation</ToolTip>
+    );
+    const selectors = REQUEST_TYPES.map((requestType, key) => {
+      const selector = (
+        <button
+          key={key}
+          value={requestType}
+          className={classNames('btn', 'btn-default', {active: this.getValue('requestType') === requestType})}
+          onClick={event => this.updateTypeButtonClick(event)}
+          disabled={this.getButtonsDisabled(requestType)}
+        >
+          {Utils.humanizeText(requestType)}
+        </button>
+      );
+      if (this.isEditing() && requestType === this.getValue('requestType')) {
+        return <OverlayTrigger placement="top" key={key} overlay={tooltip}>{selector}</OverlayTrigger>;
+      } else {
+        return selector;
+      }
+    })
+    return <div className="btn-group">{selectors}</div>;
+  }
+
+  renderLoadBalanced() {
+    const checkbox = (
+      <label htmlFor="load-balanced" className={classNames({subtle: this.isEditing()})}>
+        <CheckBox
+          id = "load-balanced"
+          onChange = {event => this.updateField("loadBalanced", !this.getValue("loadBalanced"))}
+          checked = {this.getValue("loadBalanced")}
+          disabled = {this.isEditing()}
+          noFormControlClass = {true}
+        />
+        {" Load balanced"}
+      </label>
+    );
+    let field;
+    if (this.isEditing()) {
+      field = (
+        <OverlayTrigger
+          placement="top"
+          overlay = {<ToolTip id="cannotChangeLoadBalancedAfterCreation">Option cannot be altered after creation</ToolTip>}>
+          {checkbox}
+        </OverlayTrigger>
+      );
+    } else {
+      field = checkbox;
+    }
+    return (
+      <div className = 'form-group'>
+        {field}
+      </div>
+    );
+  }
+
+  renderRequestTypeSpecificFormFields() {
```

Comment:
Maybe move some of these form definitions out to their own variables just so you can see how this function is rendering on a top-level without seeing the business code for what each field looks like?